test/loco/integer/sendmoremoney1.clj

```diff
@@ -0,0 +1,124 @@
+(ns ^{:author "Andrea Richiardi"
+      :doc "Send + More = Money example + tests." }
+  loco.integer.sendmoremoney1
+  (:refer-clojure :exclude [send]) 
+  (:require [clojure.test :refer :all]
+            [loco.core :refer :all]
+            [loco.constraints :refer :all]
+            [criterium.core :refer [bench]]))
+
+
+(defn costraint-0-9 [vars]
+  (for [v vars]
+    ($in v 0 9))) 
+
+(def send
+  (for [l "send"]
+    [:send l])) 
+
+(def more
+  (for [l "more"]
+    [:more l])) 
+
+(def money
+  (for [l "money"]
+    [:money l])) 
+
+(def letter-ties
+  [ ;; letter e
+   ($= [:send \e] [:more \e]) 
+   ($= [:send \e] [:money \e]) 
+   ($= [:more \e] [:money \e]) 
+   ;; letter o
+   ($= [:more \o] [:money \o]) 
+   ;; Letter m
+   ($= [:more \m] [:money \m])
+   ;; Letter n
+   ($= [:send \n] [:money \n])]) 
+
+(def carries
+  (for [i (range 4)]
+    ($in [:_c i] 0 1))) 
+
+(def ^{:doc "Naive model. No $distinct on each letter but on each word, which yields multiple
+ solutions. Constraint on the sum with carries:
+               c3 c2 c1 c0     
+                S  E  N  D +
+                M  O  R  E =
+             ---------------
+             M  O  N  E  Y  "}
+  smm-naive-model
+  (concat (costraint-0-9 send)
+          (costraint-0-9 more)
+          (costraint-0-9 money)
+          letter-ties
+          carries
+          [($distinct send)
```

Comment:
I believe your current model does not properly represent the problem, but the unified `$distinct` I'm suggesting would actually improve the model (and hopefully get it down to one solution). Therefore I'd like you to change it here so the constraint program is correct.